x-pack/plugins/apm/server/lib/helpers/fetch_with_timeout/index.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import fetch from 'node-fetch';
+import { ErrorWithStatusCode } from './error_with_status_code';
+
+export const fetchWithTimeout = async (
```

Comment:
Do we still need this, now that it is configurable? If we do need it, I suggest to move it to a separate endpoint so the call to get the agent statistics does not block for 5 seconds.

Reply:
At which point we probably don't need a custom timeout either?

Reply:
I think we still need this function, specially when there is no configuration set, but it's a good idea to move it to a separate endpoint so we don't block the response in statistics.

Reply:
WDYM with "when there is no configuration set"? That people are using the defaults but haven't allowed network access?

Reply:
yes, when they are using defaults but don't have access

Reply:
What happens if we just use the normal fetch? I assume it also times out eventually?

Reply:
I created a dedicated route for latest agent versions in [d5d4301](https://github.com/elastic/kibana/pull/153643/commits/d5d4301a544111cac2226dec7cdcd7c133eb39ed), also deleted the special `fetch`.